uui-core/src/data/processing/views/Tree.ts

```diff
@@ -223,47 +215,71 @@ export class Tree<TItem, TId> {
                 parentSubtotals = add(itemSubtotals, subtotalsMap.get(parentId));
             }
             subtotalsMap.set(parentId, parentSubtotals);
-        }, { direction: 'bottom-up' })
+        }, { direction: 'bottom-up' });
         return subtotalsMap;
     }
 
     public isFlatList() {
         return this.byParentId.size <= 1;
     }
 
-    public append(itemsToAdd: TItem[]) {
-        if (!itemsToAdd || itemsToAdd.length === 0) {
+    public patch(
+        items: TItem[],
+        isDeletedProp?: keyof TItem,
+        comparator?: (newItem: TItem, existingItem: TItem) => number,
+    ) {
+        if (!items || items.length === 0) {
             return this;
         }
 
         const newById = this.cloneMap(this.byId);
         const newByParentId = this.cloneMap(this.byParentId); // shallow clone, still need to copy arrays inside!
 
-        itemsToAdd.forEach((item) => {
+        let isPatched = false;
+        items.forEach((item) => {
             const id = this.getId(item);
             const existingItem = this.byId.get(id);
+            const parentId = this.getParentId(item);
+            let list = [...(newByParentId.get(parentId) ?? [])];
```

Comment:
Probably we can avoid cloning the list for every item, even if it's not inserted/deleted?

Reply:
Done.